src/Components/Components/src/Routing/Router.cs

```diff
@@ -56,6 +56,8 @@ public class Router : IComponent, IHandleAfterRender, IDisposable
         /// </summary>
         [Parameter] public RenderFragment<RouteData> Found { get; set; }
 
+        [Parameter] public Func<string, List<string>> OnNavigate { get; set; }
```

Comment:
Have you consider exposing this as a feature of the NavigationManager instead? I can think of exposing a method on WebAssemblyHostBuilder.

Reply:
I liked the idea of exposing the `OnNavigate` property as a configuration on the router because, from a developer experience, it feels a little bit more natural to have all the logic that defines route -> assembly mappings sit closer to where the routes are defined in the Razor pages. It "feels" more intuitive to have it here.

Alternatively, maybe instead of using this delegate pattern we can explore the option we discussed during design reviews of reading from a JSON-based configuration file.

That would be an interesting option to explore because it would allow devs to extend on the lazy-load feature by generating the configuration based on whatever semantics they have. We'd just be responsible for reading and processing the assemblies.

I'll tinker with this idea and push a commit.